pkg/envoy/envoy.go

```diff
@@ -150,18 +150,34 @@ func StartEnvoy(adminPort uint32, stateDir, logPath string, baseID uint64) *Envo
 	// case no one reader reads it.
 	started := make(chan bool, 1)
 	go func() {
-		logger := &lumberjack.Logger{
-			Filename:   logPath,
-			MaxSize:    100, // megabytes
-			MaxBackups: 3,
-			MaxAge:     28,   //days
-			Compress:   true, // disabled by default
+		var logWriter io.Writer
+		var logFormat string
+		if logPath != "" {
+			// Use the Envoy default log format when logging to a separate file
+			logFormat = "[%Y-%m-%d %T.%e][%t][%l][%n] %v"
+			logger := &lumberjack.Logger{
+				Filename:   logPath,
+				MaxSize:    100, // megabytes
+				MaxBackups: 3,
+				MaxAge:     28,   //days
+				Compress:   true, // disabled by default
+			}
+			defer logger.Close()
+			logWriter = logger
+		} else {
+			// Use log format that looks like Cilium logs when integrating logs
+			// The logs will be reported as coming from the cilium-agent, so
+			// we add the thread id to be able to differentiate between Envoy's
+			// main and worker threads.
+			logFormat = "level=%l msg=\"%v\" subsys=envoy-%n thread=%t"
```

Comment:
Can you confirm that %v properly quotes the string, esp. transforms `“` into `\”`?

Reply:
It does not, e.g.:
```
[2018-03-16 19:54:15.118][30623][debug][config] bazel-out/k8-fastbuild/bin/external/envoy/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:59] gRPC config for type.googleapis.com/cilium.NetworkPolicy accepted with 10 resources: [name: "f00d::a0b:0:0:dc06"
policy: 43065
...
, name: "f00d::a0b:0:0:367d"
policy: 21941
]
```